client/bio-webclient/src/main/kotlin/ac/uk/ebi/biostd/client/api/ProjectClient.kt

```diff
@@ -22,14 +24,30 @@ class ProjectClient(
 ) : ProjectOperations {
     override fun submitProject(project: File): ResponseEntity<Submission> {
         val headers = HttpHeaders().apply { MediaType.MULTIPART_FORM_DATA }
-        val body = getMultipartBody(FileSystemResource(project))
+        val body = LinkedMultiValueMap<String, Any>().apply {
+            add(PROJECT, FileSystemResource(project))
+        }
 
         return template
             .postForEntity<String>("/projects", HttpEntity(body, headers))
             .map { serializationService.deserializeSubmission(it, SubFormat.JSON_PRETTY) }
     }
 
-    private fun getMultipartBody(project: Any) = LinkedMultiValueMap<String, Any>().apply {
-        add(PROJECT, project)
+    override fun attachSubmission(
+        projectAccNo: String,
+        submission: File,
+        files: List<File>
+    ): ResponseEntity<Submission> {
+        val headers = HttpHeaders().apply { MediaType.MULTIPART_FORM_DATA }
+        val body = getMultipartBody(files, FileSystemResource(submission))
+
+        return template
+            .postForEntity<String>("/projects/$projectAccNo/submissions", HttpEntity(body, headers))
+            .map { serializationService.deserializeSubmission(it, SubFormat.JSON_PRETTY) }
+    }
+
+    private fun getMultipartBody(files: List<File>, submission: Any) = LinkedMultiValueMap<String, Any>().apply {
```

Comment:
IMO always favor construction over mutation. 

`files.map { FILES to FileSystemResource(it) }.plus { SUBMISSION to FileSystemResource(submission) }.toMap()`

In this way you can use not a mutable object map. In general. In general we should never use mutable data structures,  is the exception because it use reflection so is mandatory.